### PR TITLE
ci: add nightly workflow

### PR DESCRIPTION
**Pull request by jsarzy**
## Description
Add nightly CI that will be run at 1 am. It includes long tests such as busybox or micropython test suit. I transformed the regular workflow into a callable one with the 'nightly' argument.

A PR from phoenix-rtos-tests is required for this, and I will handle the merging myself.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: ia32-generic.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: nightly
+
+on:
+  schedule:
+    - cron: '0 1 * * *'
+  workflow_dispatch:
+
+jobs:
+  call-ci:
+    uses: phoenix-rtos/phoenix-rtos-project/.github/workflows/ci-project.yml@jsarzy/nightly
```
Comment (jsarzy) [inline .github/workflows/nightly.yml:10]: must be changed to master before merging

**Comment by jsarzy** [comment]
Must wait for https://github.com/phoenix-rtos/phoenix-rtos-tests/pull/200

["nightly" workflow triggered manually](https://github.com/phoenix-rtos/phoenix-rtos-project/actions/runs/5002669885) (it's expected to fail)

**Review by damianloew — CHANGES_REQUESTED** [review]
Looks good to me, I'd like to see only the current results (for example triggered on `pull_request` event - if everything was ok we could change branches to master and merge it)

File: .github/actions/phoenix-runner/action.yml [hunk 2]
```diff
@@ -11,15 +11,23 @@ inputs:
...
-
+    using: 'docker'
+    image: 'phoenixrtos/devel'
+    entrypoint: ./phoenix-rtos-tests/runner.py
+    args:
+      - '-T${{ inputs.target }}'
+      # HACK: We can't pass an empty argument to the runner due to strict arg parsing.
+      # Additionally, there is no conditional arg in github actions. If this is not a nightly run
+      # then repeating the target flag is acceptable, the runner is OK with that.
+      - ${{ inputs.nightly == 'true' && '--nightly' || format('-T{0}', inputs.target) }}
```
[the remark this answers is not in the document]
Comment (jsarzy) [reply .github/actions/phoenix-runner/action.yml:30]: I think the best solution here is to use `argpasrse.BooleanOptionalAction` in the python script. With that you can just pass `--no-nightly` flag. However, this requires python >= 3.9.

**Review by nalajcie — APPROVED** [review]

**Pull request merged**